### PR TITLE
OCM-7378 | ci: Bypass the tags setting in case 73068 due to OCM-7341

### PR DESCRIPTION
**Pull request by xueli181114**
**What this PR does / why we need it**:
Bypass the tags setting in case 73068 due to [OCM-7341](https://issues.redhat.com//browse/OCM-7341)
**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
Fixes #[OCM-7378](https://issues.redhat.com//browse/OCM-7378)

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [ ] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

**Comment by xueli181114** [comment]
`lixue@Xue-Lis-MacBook-Pro terraform-provider-rhcs % ginkgo -focus 73068 tests/e2e
time="2024-04-11T15:36:25+08:00" level=info msg="Running against env staging with gateway url https://api.stage.openshift.com"
time="2024-04-11T15:36:25+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path: version: version_pattern:latest zones:a,b,c]"
Running Suite: e2e tests suite - /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e
================================================================================================
Random Seed: 1712820980

Will run 1 of 71 specs
time="2024-04-11T15:36:25+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path: version: version_pattern:latest zones:a,b,c]"
time="2024-04-11T15:36:25+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
time="2024-04-11T15:36:29+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSStime="2024-04-11T15:36:31+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
time="2024-04-11T15:36:32+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/vpc"
time="2024-04-11T15:36:36+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/vpc"
time="2024-04-11T15:36:38+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/security-groups"
time="2024-04-11T15:36:42+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/security-groups"
time="2024-04-11T15:36:44+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/security-groups"
time="2024-04-11T15:36:47+08:00" level=info msg="Running terraform apply against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var name=ocp-73068 -var url=https://api.stage.openshift.com -var replicas=0 -var autoscaling_enabled=false -var auto_repair=false -var tags={\"tag1\":\"test_tag1\",\"tag2\":\"test_tag2\"} -var cluster=2aii307hd4eijdiphd5t7kphf9hpjgbn -var machine_type=r5.xlarge -var subnet_id=subnet-02adf8252306aee2d -var additional_security_groups=[\"sg-0392263b6caae66a9\",\"sg-08717a5f4546c7e6d\",\"sg-01b0735b9ad364550\",\"sg-0316e39bb424296a9\"]]"
time="2024-04-11T15:36:53+08:00" level=info msg="Recording tfvars file /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp/terraform.tfvars"
time="2024-04-11T15:36:53+08:00" level=info msg="Running terraform apply against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var additional_security_groups=[\"sg-0392263b6caae66a9\"] -var url=https://api.stage.openshift.com]"
time="2024-04-11T15:36:56+08:00" level=error msg="rhcs_hcp_machine_pool.mp: Refreshing state... [id=ocp-73068]\n\nTerraform used the selected providers to generate the following execution\nplan. Resource actions are indicated with the following symbols:\n  ~ update in-place\n\nTerraform will perform the following actions:\n\n  # rhcs_hcp_machine_pool.mp will be updated in-place\n  ~ resource \"rhcs_hcp_machine_pool\" \"mp\" {\n      ~ aws_node_pool     = {\n          ~ additional_security_group_ids = [\n                \"sg-0392263b6caae66a9\",\n              - \"sg-08717a5f4546c7e6d\",\n              - \"sg-01b0735b9ad364550\",\n              - \"sg-0316e39bb424296a9\",\n            ]\n            tags                          = {\n                \"tag1\" = \"test_tag1\"\n                \"tag2\" = \"test_tag2\"\n            }\n            # (2 unchanged attributes hidden)\n        }\n        id                = \"ocp-73068\"\n        name              = \"ocp-73068\"\n      ~ status            = {\n          ~ current_replicas = 0 -> (known after apply)\n          ~ message          = \"WaitingForAvailableMachines\" -> (known after apply)\n        } -> (known after apply)\n        # (7 unchanged attributes hidden)\n    }\n\nPlan: 0 to add, 1 to change, 0 to destroy.\nrhcs_hcp_machine_pool.mp: Modifying... [id=ocp-73068]\n\nError: Attribute value cannot be changed\n\n  with rhcs_hcp_machine_pool.mp,\n  on main.tf line 26, in resource \"rhcs_hcp_machine_pool\" \"mp\":\n  26: resource \"rhcs_hcp_machine_pool\" \"mp\" {\n\nAttribute aws_node_pool.additional_security_group_ids, cannot be changed from\n[\"sg-0392263b6caae66a9\",\"sg-08717a5f4546c7e6d\",\"sg-01b0735b9ad364550\",\"sg-0316e39bb424296a9\"]\nto [\"sg-0392263b6caae66a9\"]"
time="2024-04-11T15:36:56+08:00" level=info msg="Running terraform destroy against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
args:  /usr/local/bin/terraform destroy -auto-approve -no-color -var url=https://api.stage.openshift.com -var additional_security_groups=["sg-0392263b6caae66a9"]
time="2024-04-11T15:37:00+08:00" level=info msg="Deleting tfvars file /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp/terraform.tfvars"
time="2024-04-11T15:37:00+08:00" level=info msg="Running terraform apply against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var machine_type=m5.2xlarge -var replicas=0 -var autoscaling_enabled=false -var cluster=2aii307hd4eijdiphd5t7kphf9hpjgbn -var name=add-73068 -var url=https://api.stage.openshift.com -var subnet_id=subnet-02adf8252306aee2d -var auto_repair=false -var tags={\"tag1\":\"test_tag1\",\"tag2\":\"test_tag2\"}]"
args:  /usr/local/bin/terraform destroy -auto-approve -no-color -var machine_type=m5.2xlarge -var replicas=0 -var subnet_id=subnet-02adf8252306aee2d -var tags={"tag1":"test_tag1","tag2":"test_tag2"} -var cluster=2aii307hd4eijdiphd5t7kphf9hpjgbn -var name=add-73068 -var auto_repair=false -var url=https://api.stage.openshift.com -var autoscaling_enabled=false
time="2024-04-11T15:37:06+08:00" level=info msg="Running terraform destroy against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
time="2024-04-11T15:37:11+08:00" level=info msg="Running terraform destroy against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
args:  /usr/local/bin/terraform destroy -auto-approve -no-color -var machine_type=m5.2xlarge -var replicas=0 -var subnet_id=subnet-02adf8252306aee2d -var tags={"tag1":"test_tag1","tag2":"test_tag2"} -var url=https://api.stage.openshift.com -var name=add-73068 -var autoscaling_enabled=false -var auto_repair=false -var cluster=2aii307hd4eijdiphd5t7kphf9hpjgbn
•SSSSSSSSSS

Ran 1 of 71 Specs in 46.128 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 70 Skipped
PASS

Ginkgo ran 1 suite in 51.338628759s
Test Suite Passed`

File: tests/e2e/hcp_machine_pool_test.go [hunk 1]
```diff
@@ -202,32 +201,43 @@ var _ = Describe("HCP MachinePool", ci.Day2, ci.NonClassicCluster, ci.FeatureMac
...
+
+			// workaround
+			By("Check cluster tags and set it to the machinepool")
+			resp, err := cms.ListNodePools(ci.RHCSConnection, clusterID)
+			Expect(err).ToNot(HaveOccurred())
+			npDetail, err := cms.RetrieveNodePool(ci.RHCSConnection, clusterID, resp[0].ID(), map[string]interface{}{
+				"fetchUserTagsOnly": true,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			workAroundTags := npDetail.AWSNodePool().Tags()
```
Comment (radtriste) [inline tests/e2e/hcp_machine_pool_test.go:238]: I guess at some point we should have a `helper.RetrieveClusterUserTags` method. wdyt ?

File: tests/e2e/hcp_machine_pool_test.go [hunk 2]
```diff
@@ -202,32 +201,43 @@ var _ = Describe("HCP MachinePool", ci.Day2, ci.NonClassicCluster, ci.FeatureMac
...
 			machineType := "r5.xlarge"
-			name := "ocp-73068-2"
+			name := "ocp-73068"
 			sgIDs := output.SGIDs
 			if len(sgIDs) >= 4 {
 				sgIDs = sgIDs[0:4]
 			}
-			MachinePoolArgs := &exe.MachinePoolArgs{
+
+			// workaround
```
Comment (radtriste) [inline tests/e2e/hcp_machine_pool_test.go:230]: Can you add the issue number ?

**Review by radtriste — APPROVED** [review]
minor comment

File: tests/e2e/hcp_machine_pool_test.go (hunk 2 above)
Comment (xueli181114) [reply tests/e2e/hcp_machine_pool_test.go:230]: According to previous discussion with @yasun1 , we are not supposed to have issue number in code.

File: tests/e2e/hcp_machine_pool_test.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/hcp_machine_pool_test.go:238]: @radtriste Make sense. But right now it's not valuable for a workaround

File: tests/e2e/hcp_machine_pool_test.go (hunk 2 above)
Comment (radtriste) [reply tests/e2e/hcp_machine_pool_test.go:230]: oki

**Pull request merged**